OOP - Milestone: game logic

bmuhamood opened the pull request:
-  Created branch name game_logic.
- Removed all inline comments
- Added ! in all areas where user interaction used. 
- Fixed all linter issues

euqueme reviewed (changes requested):
> ### Pull Request reviewed by Maria Eugenia Queme P. (TSE)

### Hello @bmuhamood and @t-yanick :balloon: :lollipop: 

### Thank you for your submission :handshake: :handshake: you have a great start here I see you've created a class and also made use of attr_accessors I have to say I had no problem running the game and no execution errors where showing, :medal_sports: :medal_sports:  lastly claps to you for having a very accomplished Readme file for this milestone :clap: :clap: :clap: having said that there is still work to do so let's get to it shall we? :construction: :construction_worker: :construction_worker: 

#### General
- [x] Files not related to the project should not be committed and added to .gitignore (e.g. `.DS_Store`) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
> let's remove the '.DS_Store' file and add it to the .gitignore so it doesn't get committed

#### Behaviour
* [ ] The application can tell if there was a draw after players move
> Your game doesn't end when a draw is detected it keeps asking players for moves, let's fix it
![image](https://user-images.githubusercontent.com/52209067/92982193-56e05d80-f45a-11ea-82e4-08bae78f7676.png)

- [ ] To start another game your game asks the player to input a `y` or an `n` but they have to be upper case, let's specify this in the instructions or change it so it doesn't matter if they are upper or down case.
![image](https://user-images.githubusercontent.com/52209067/92982298-c5252000-f45a-11ea-902f-72d437b1879b.png)

#### Implementation details

* [ ] The project uses at least two classes (Usually it uses three: Game, Board, and Player). The code of the project should be logically divided between classes. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
> I see you have the same class in both your `bin/main.rb` file and your `lib/game_logic.rb` file, all classes should be put in the lib directory and the name of the file that holds the class should have the same name of the class (ie. the name of your class is Tic it should be in a file called tic.rb).

* [ ] All game-logic related resources (classes and/or modules) are kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
> the `bin/main.rb` file shouldn't be a class it should only contain the game loop and have objects of the classes created in the `lib` directory.

* [ ] Only use attr_accessors if you really need it  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))
> your `:player` and `:player1` `attr_accessors` could be changed to `attr_readers` and your game still runs the same so make sure you use `attr_accessors` where they are really needed.

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any questions? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :traffic_light:

bmuhamood commented:
- Added DS_Store' file and add it to the .gitignore
- The new game can now detect draw
- Removed player to input a y or an n option
- Created new different game classes as requested in code review
- All game-logic related resources (classes and/or modules) are kept in the lib directory.
- Changed as requested to **attr_reader**
- Fixed all linters issues
- Fixed all game logics and errors

expjazz commented on bin/main:
```diff
@@ -1,104 +1,72 @@
-#!/usr/bin/env ruby
```
- [ ] You need to leave the command of the image below to execute.

![image](https://user-images.githubusercontent.com/61479648/93138862-1e3cc000-f6b6-11ea-85e3-63904e1c1c34.png)

expjazz reviewed (changes requested):
## Status: Changes Required :recycle: 

Hey :+1: 

You did a great job here, and almost everything is working perfectly. However, there are a few details to improve. 

- [ ] The game does not execute anymore. Check the inline comments for more detail.

- [ ] There is a bug with the names. The player two is starting, and the print statements are always printing "player one turn". Can you make this message more clear? And maybe not allow the user to put numbers as players' names.

![image](https://user-images.githubusercontent.com/61479648/93139265-cd799700-f6b6-11ea-8661-98566bf6ee40.png)

#### As stated above, the project looks really good.

### Please make the required changes and resubmit for a new code review once you are ready
Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

bmuhamood commented:
- Fixed all the requested bugs and meaningful messages. 
- Added the requested line of command.
- Fixed all the linters and code is working fine.

expjazz reviewed (approved):
## Status: Project Approved :clap: 

### Great job on the changes and on the project. Awesome :1st_place_medal: 

Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_
